spec/rubocop/cop/chef/modernize/respond_to_provides_spec.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright:: 2019, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'spec_helper'
+
+describe RuboCop::Cop::Chef::RespondToProvides, :config do
+  subject(:cop) { described_class.new(config) }
+
+  it 'registers an offense with a HWRP that uses respond_to? with provides' do
+    expect_violation(<<-RUBY)
+    class Chef
+      class Provider
+        class Icinga2Environment < Chef::Provider::LWRPBase
+          provides :icinga2_environment if respond_to?(:provides)
+          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ respond_to?(:provides) in resources is no longer necessary in Chef Infra Client 12+
```

Comment:
shouldn't you target only the `if respond_to?(:provides)` bit?

Reply:
I'm not entirely sure how to do that to be honest. 

Here's the ast of the above test:
```ruby
(if
  (send nil :respond_to?
    (sym :provides))
  (send nil :provides
    (sym :icinga2_environment)) nil)
```

We need to match the if respond_to?(:provides), but you can't do a selector that doesn't select on the child node of (sym :icinga2_environment) as far as I know. That's why I matches the whole node and manipulated it in the autocorrect.